File: javascript/src/dominio/Sessao.js
// Sessao autenticada. Da uso concreto ao email e a senha do Usuario.
export class Sessao {
  #token;
  #usuarioId;
  #expiraEm;

  constructor(token, usuarioId, expiraEm) {
    this.#token = token;
    this.#usuarioId = usuarioId;
    this.#expiraEm = expiraEm;
  }

  estaValida() {
    return new Date() < this.#expiraEm;
  }

  get token() { return this.#token; }
  get usuarioId() { return this.#usuarioId; }
  get expiraEm() { return this.#expiraEm; }
}

File: java/src/br/unifor/facilitador/dominio/Sessao.java
package br.unifor.facilitador.dominio;

import java.time.LocalDateTime;
import java.util.UUID;

/** Sessao autenticada. Da uso concreto ao email e a senha do Usuario. */
public class Sessao {

    private final String token;
    private final UUID usuarioId;
    private final LocalDateTime expiraEm;

    public Sessao(String token, UUID usuarioId, LocalDateTime expiraEm) {
        this.token = token;
        this.usuarioId = usuarioId;
        this.expiraEm = expiraEm;
    }

    public boolean estaValida() {
        return LocalDateTime.now().isBefore(expiraEm);
    }

    public String getToken() { return token; }
    public UUID getUsuarioId() { return usuarioId; }
    public LocalDateTime getExpiraEm() { return expiraEm; }
}
